iset.mm

```diff
@@ -61226,6 +61246,23 @@ would need to be strengthened (for example, to imply included middle) to
       WMVPRXEVQVRVSVTVTWJWAWB $.
   $}
 
+  ${
+    $d A w y z $.  $d B w y z $.
+    $( The union of two disjoint finite sets is finite.  (Contributed by Jim
+       Kingdon, 25-Feb-2022.) $)
+    unfidisj $p |- ( ( A e. Fin /\ B e. Fin /\ ( A i^i B ) = (/) )
+        -> ( A u. B ) e. Fin ) $=
```

Comment:
Why is ( A i^i B ) = (/) necessary? Is ( ( A e. Fin /\ B e. Fin ) -> ( A u. B ) e. Fin ) not provable (see ~unfi in set.mm).

Reply:
unfi is not provable. As we say in http://us.metamath.org/ileuni/mmil.html , "Not provable according to Remark 8.1.17 of [AczelRathjen]"

Although we don't have a specific proof that unfi implies excluded middle, I assume it does and that a proof along the lines of http://us.metamath.org/ileuni/ssfiexmid.html or http://us.metamath.org/ileuni/domfiexmid.html would work.

Reply:
OK, thanks a lot for the explanation.

Reply:
Let `X = { x e. 1o | ph }` be my favorite ordinal. Then `{ 1o } e. Fin` and `{ X } e. Fin` but if `( { 1o } u. { X } ) e. Fin` then either this set has cardinality 1 and `X = 1o` (so `ph`), or the set has cardinality 2 and `X =/= 1o` (hence `-. ph`, since `ph` implies `X = 1o`).

Reply:
A stronger version of the theorem that may be easier to apply in practice is `( ( A e. Fin /\ B e. Fin /\ ( A i^i B ) e. Fin ) -> ( A u. B ) e. Fin )`. I think that's even a biconditional: `( ( A e. Fin /\ B e. Fin ) -> ( ( A u. B ) e. Fin <-> ( A i^i B ) e. Fin ) )`. Another variation that looks helpful in proving this is `( ( A e. Fin /\ B e. Fin /\ B C_ A ) -> A = ( B u. ( A \ B ) ) )` (and `A\B` is also finite)

Reply:
Ooh, nice, I'll make a separate pull request for "unfi implies excluded middle" (which formalizes pretty much exactly as stated above) and probably for the others too.

Reply:
This seems to be going well but since it probably won't be done today I'll put the progress so far in a draft pull request at #2536